pywinauto/base_wrapper.py

```diff
@@ -927,9 +910,47 @@ def type_keys(
         self.actions.log('Typed text to the ' + friendly_class_name + ': ' + aligned_keys)
         return self
 
-    #-----------------------------------------------------------
+    # -----------------------------------------------------------
+    def type_keys(
+            self,
+            keys,
+            pause=None,
+            with_spaces=False,
+            with_tabs=False,
+            with_newlines=False,
+            turn_off_numlock=True,
+            set_foreground=True):
+        """
+          Type keys to the element using keyboard.SendKeys
+
+          This uses the re-written keyboard_ python module where you can
+          find documentation on what to use for the **keys**.
+
+          .. _keyboard: pywinauto.keyboard.html
+          """
+        return self._type_keys(keys, keyboard.SendKeys, pause=pause, with_spaces=with_spaces, with_tabs=with_tabs,
+                               with_newlines=with_newlines,
+                               turn_off_numlock=turn_off_numlock, set_foreground=set_foreground)
+
+    # -----------------------------------------------------------
+    def key_up(self, key, set_foreground=True):
```

Comment:
I don't think special methods for single key down or up is the only way. Method `type_keys` should provide some notation like `.type_keys("{ALT down}123{ALT up}{SHIFT down}hello{SHIFT up}"`. For single key it may look like `{h down}{i down}{h up}{i up}`.

Reply:
So you want type_keys to handle single key strokes (both keyup and keydown), the way you mentioned above instead of having separate methods?

Reply:
Yes, I'm not sure about separate methods. Good API should be a little bit redundant, but 10 calls to one method in the test code looks not good.